packages/react-core/src/components/TextArea/TextArea.tsx

```diff
@@ -11,15 +11,17 @@ export enum TextAreResizeOrientation {
   both = 'both'
 }
 
-export interface TextAreaProps extends Omit<HTMLProps<HTMLTextAreaElement>, 'onChange' | 'ref'> {
+export interface TextAreaProps extends Omit<HTMLProps<HTMLTextAreaElement>, 'onChange' | 'ref' | 'readOnly'> {
   /** Additional classes added to the TextArea. */
   className?: string;
   /** Flag to show if the TextArea is required. */
   isRequired?: boolean;
   /** Flag to show if the TextArea is disabled. */
   isDisabled?: boolean;
-  /** Flag to show if the TextArea is read only. */
+  /** @deprecated Use readOnly instead. Flag to show if the TextArea is read only. */
   isReadOnly?: boolean;
+  /** Read only variant. */
+  readOnly?: 'default' | 'plain';
```

Comment:
Nit: I would suggest something along the lines of `readOnlyVariant`, just to ensure the prop isn't misconstrued as taking a boolean value. If updated this would also apply to the text input tsx file as well.

Reply:
Great point @thatblindgeye  ... I like `readOnlyVariant`! Will update accordingly